app/src/main-process/main.ts

```diff
@@ -147,29 +89,37 @@ function handleAppURL(url: string) {
   })
 }
 
-const isDuplicateInstance = app.makeSingleInstance((args, workingDirectory) => {
-  // Someone tried to run a second instance, we should focus our window.
-  if (mainWindow) {
-    if (mainWindow.isMinimized()) {
-      mainWindow.restore()
-    }
+let isDuplicateInstance = false
+// If we're handling a Squirrel event we don't wanna enforce single instance. We
+// want to let the updated instance launch and do its work. It will then quit
+// once it's done.
+if (!handlingSquirrelEvent) {
+  isDuplicateInstance = app.makeSingleInstance((args, workingDirectory) => {
+    // Someone tried to run a second instance, we should focus our window.
+    if (mainWindow) {
+      if (mainWindow.isMinimized()) {
+        mainWindow.restore()
+      }
+
+      if (!mainWindow.isVisible()) {
+        mainWindow.show()
+      }
 
-    if (!mainWindow.isVisible()) {
-      mainWindow.show()
+      mainWindow.focus()
     }
 
-    mainWindow.focus()
-  }
+    if (args.length > 1) {
+      handleAppURL(args[1])
+    }
+  })
 
-  if (args.length > 1) {
-    handleAppURL(args[1])
+  if (isDuplicateInstance) {
+    app.quit()
   }
-})
-
-if (isDuplicateInstance) {
-  willQuit = true
+}
 
-  app.quit()
+if (shellNeedsPatching(process)) {
```

Comment:
I'm curious (but not really surprised) about moving the patching after all the squirrel checks. Just playing it safe?